test/test_other.py

```diff
@@ -5942,11 +5942,19 @@ def test_force_stdlibs(self):
     self.do_runf('hello_world.c', emcc_args=['-sWASM_BIGINT'])
 
   @crossplatform
-  @also_with_env_modify({'gb_locale': {'LC_ALL': 'en_GB'}, 'long_tz': {'TZ': 'Asia/Kathmandu'}})
   def test_strftime_zZ(self):
-    if os.environ.get('LC_ALL') == 'en_GB' and MACOS:
+    if MACOS:
       self.skipTest('setting LC_ALL is not compatible with macOS python')
-    self.do_runf('other/test_strftime_zZ.c', 'ok!')
+
+    tz_lang_envs = [
+        {"LC_ALL": "en_GB", "TZ": "Europe/London"},
+        {"LC_ALL": "th_TH", "TZ": "Asia/Bangkok"},
+        {"LC_ALL": "ar-AE", "TZ": "United Arab Emirates"},
```

Comment:
Can you add to the PR description explaining what these result in both before and after this change?

Reply:
I've udpated the PR description with various examples of failed processing along with the new correct values :)